src/authentication/DemoSiteRequest.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+import { hashPassword } from 'sussol-utilities';
+export class DemoSiteRequest {
+  async createActivationURL(username, email, password, repeatPassword) {
+    // Client side validation
+    if (email.length === 0) throw new Error('Enter the E-mail');
+    if (!this.validateEmail(email)) throw new Error('Enter a valid E-mail');
+    if (username.length === 0) throw new Error('Enter the username');
+    if (password.length === 0) throw new Error('Enter the password');
+    if (repeatPassword.length === 0) throw new Error('Enter the repeat password');
+    if (password !== repeatPassword) throw new Error('Password & repeat password must match');
+    if (this.textLengthBetween(password)) throw new Error('Password must be 8-32 characters long');
```

Comment:
the names of the methods threw me at first - the method is 'textLengthBetween' and it returns true if the text length is not between the values. 'textNotBlank' has the logic the other way - it returns true if the test is ok.

The names could be 
if (this.textLengthInvalid(password))...
if (this.textContainsSpaces(password))...

Reply:
Please check https://github.com/sussol/mobile/pull/724/commits/185e242891f94982fe9b887bab41f42624d53066